File: 4.1_First_Fragment_Project/app/src/main/java/com/example/firstfragmentproject/MainActivity.java
package com.example.firstfragmentproject;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.fragment.app.FragmentActivity;

public class MainActivity extends FragmentActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        if (savedInstanceState == null) {
            getSupportFragmentManager()
                    .beginTransaction()
                    .add(R.id.fragment_container, FirstFragment.class, null)
                    .commit();
        }
    }

    @Override
    protected void onSaveInstanceState(@NonNull Bundle outState) {
        super.onSaveInstanceState(outState);
    }
}

File: 9.6_WeatherApplication/app/src/main/java/com/example/a96_weatherapplication/screen/MainActivity.kt
package com.example.a96_weatherapplication.screen

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.example.a96_weatherapplication.R
import com.example.a96_weatherapplication.screen.forecastlist.ForecastListFragment

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val forecastListFragment = ForecastListFragment();
        val manager = supportFragmentManager
        manager.beginTransaction()
            .replace(R.id.fragment_container, forecastListFragment)
            .setReorderingAllowed(true)
            .addToBackStack(null)
            .commit()
    }
}